python/beeai_framework/agents/tool_calling/agent.py

```diff
@@ -192,3 +196,22 @@ def memory(self) -> BaseMemory:
     @memory.setter
     def memory(self, memory: BaseMemory) -> None:
         self._memory = memory
+
+    @staticmethod
+    def _generate_templates(
+        overrides: dict[ToolCallingAgentTemplatesKeys, PromptTemplate[Any] | ToolCallingAgentTemplateFactory]
+        | None = None,
+    ) -> ToolCallingAgentTemplates:
+        templates = ToolCallingAgentTemplates()
+        if overrides is None:
+            return templates
+
+        for name, _info in templates.model_fields.items():
+            override: PromptTemplate[Any] | ToolCallingAgentTemplateFactory | None = overrides.get(name)
+            if override is None:
+                continue
+            elif isinstance(override, PromptTemplate):
+                setattr(templates, name, override)
+            else:
+                setattr(templates, name, override(getattr(templates, name)))
```

Comment:
Not sure whether using `setattr` and `getatter` + `override` is the best way to do it.

Reply:
I thought the same, but every time I thought of a different idea it either didn't work or google suggested I use setattr and getatter. I'm open to another solution here if you have one